src/Splash/Splash.Animation.tsx

```diff
@@ -34,73 +36,75 @@ const bounce = (coord: number): number => {
   return coord
 }
 
-const Container = styled("div")<{ size: number }>(({ size }: { size: number }) => ({
-  width: size,
-  height: size,
+const Container = styled("div")({
   position: "absolute",
   top: "50%",
   left: "50%",
   transform: "translate3d(-50%, -50%, 0)",
-}))
+})
+
+/// Move highly highly dynamic style out of css-js to prevent uneeded classname generation
 
-const Box = styled("div")<{ x: number; y: number }>(({ x, y }) => ({
+const Box = styled("div")({
   position: "absolute",
   transition: "all 0.5s ease-in-out",
-  top: `calc(${(x / (squares - 1)) * 100}% + 4px)`,
-  left: `calc(${(y / (squares - 1)) * 100}% + 4px)`,
+
   borderRadius: 6,
   width: `calc(${100 / (squares - 1)}% - 8px)`,
   height: `calc(${100 / (squares - 1)}% - 8px)`,
   backgroundColor: "rgba(255, 255, 255, 0.06)",
-}))
-
-class Animation extends React.Component<Props, State> {
-  public state = {
-    animationStep: 0,
-    coordinates: Array.apply(null, { length: boxes })
-      .map(Number.call, Number)
-      .map(() => ({ x: integerRandom(squares), y: integerRandom(squares) })),
-  }
+})
 
-  public animationInterval?: number
-
-  // Shift the coordinate of every third tile in a random direction.
-  // Each animation shifts a different set of tiles.
-  public shiftSomeTiles() {
-    this.setState(prevState => ({
-      animationStep: prevState.animationStep + 1,
-      coordinates: prevState.coordinates.map((coord: { x: number; y: number }, index: number) => {
-        if (index % 3 === prevState.animationStep % 3) {
-          const dx = integerRandom(3) - 1
-          const dy = integerRandom(3) - 1
-          return {
-            x: bounce(coord.x + dx),
-            y: bounce(coord.y - dy),
-          }
-        }
-        return coord
-      }),
-    }))
-  }
+const initialState = {
+  animationStep: 0,
+  coordinates: Array.apply(null, Array(boxes))
+    .map((_: any, i: number) => i) // Fixes typescript Error
+    .map(() => ({ x: integerRandom(squares), y: integerRandom(squares) })),
+}
 
-  public componentDidMount() {
-    this.animationInterval = window.setInterval(this.shiftSomeTiles.bind(this), 5000)
-  }
+const Animation: React.FC<Props> = ({ fullscreen, size }) => {
+  let containerSize = size || 600
 
-  public componentWillUnmount() {
-    clearInterval(this.animationInterval)
+  if (fullscreen) {
```

Comment:
It's a nicer naming convention to prefix booleans with `is`:

```suggestion
  if (isFullscreen) {
```

Reply:
@TejasQ - will change name, however-
This is a big mistake on my part since Hooks should not be conditional. I think i will have to initiate `useWindowSize`, and simply create a handler `const getWidth=()=> isFullscreen? Math.max(windowSize.width, windowSize.height): (size | 500)`

We should add a tslint rules, tslint-react currently has open issue [Add linting for React Hooks](https://github.com/palantir/tslint-react/issues/186) with a [PR](https://github.com/palantir/tslint-react/pull/204) ready to go.